pro/src/components/AddressManual/AddressManual.tsx

```diff
@@ -0,0 +1,90 @@
+import { useField, useFormikContext } from 'formik'
+import React from 'react'
+
+import { Callout } from 'components/Callout/Callout'
+import { CalloutVariant } from 'components/Callout/types'
+import { FormLayout } from 'components/FormLayout/FormLayout'
+import { type IndividualOfferFormValues } from 'components/IndividualOfferForm/types'
+import { ButtonLink } from 'ui-kit/Button/ButtonLink'
+import { ButtonVariant } from 'ui-kit/Button/types'
+import { TextInput } from 'ui-kit/form/TextInput/TextInput'
+import { getCoordsType, parseDms } from 'utils/coords'
+
+import styles from './AddressManual.module.scss'
+
+export const AddressManual = (): JSX.Element => {
+  const [coords, coordsMeta] = useField<string>('coords')
+
+  const formik = useFormikContext<IndividualOfferFormValues>()
+
+  const onCoordsBlur = async (event: React.ChangeEvent<HTMLInputElement>) => {
+    formik.handleChange(event)
+    await formik.setFieldTouched('coords', true, true)
+
+    const newCoords = event.target.value
+    let latitude = '',
+      longitude = ''
+
+    if (!coordsMeta.error) {
+      const coordsType = getCoordsType(newCoords)
+
+      if (coordsType === 'DD') {
+        ;[latitude, longitude] = newCoords.split(', ')
+      } else if (coordsType === 'DMS') {
+        ;[latitude, longitude] = newCoords
+          .split(' ')
+          .map((c) => String(parseDms(c)))
+      }
+    }
```

Comment:
Tu peux carrément faire ça, non ?

```suggestion
    const coordsType = getCoordsType(newCoords)

      if (coordsType === 'DD') {
        ;[latitude, longitude] = newCoords.split(', ')
      } else if (coordsType === 'DMS') {
        ;[latitude, longitude] = newCoords
          .split(' ')
          .map((c) => String(parseDms(c)))
      }
```

Reply:
(et donc virer les meta)

Reply:
En effet cependant `coordsType` utilise en interne une regex qui est une opération coûteuse. Le fait d'ajouter une condition avant permet de s'assurer qu'on exécute la regex uniquement sur une valeur dont on est sûr qu'elle est une "DD" ou "DMS"

Cela étant, je peux aussi modifier `getCoordsType` pour le faire en interne :
```tsx
export const getCoordsType = (coords: string): 'DMS' | 'DD' | 'unknown' => {
  if (!coords) return 'unknown'

  …
}```

Qu'en penses-tu ?

Reply:
Yep, ça me va !

Reply:
Done